pegasus/sites.v3/code.org/generate_pdfs.rake

```diff
@@ -0,0 +1,46 @@
+require_relative '../../src/env'
+require 'cdo/hip_chat'
+require 'cdo/rake_utils'
+require 'cdo/tempfile'
+require 'pdf/conversion'
+require src_dir 'course'
+
+PDFConversionInfo = Struct.new(:url_path, :src_files, :output_pdf_path)
+
+def pdf_conversions_for_files(file_pattern, url_extension)
+  [].tap do |conversion_infos|
+    Dir.glob(file_pattern).each do |file|
+      file_path_without_makepdf_extension = file[0...-(File.extname(file).length)]
+      file_path_without_extension = file_path_without_makepdf_extension[0...-(File.extname(file_path_without_makepdf_extension).length)]
```

Comment:
fwiw (I realize this is mostly copy and paste from the curriculum pdf script), I think it would be cleaner to use <a href="http://ruby-doc.org/core-2.2.0/File.html#method-c-basename
">basename</a> instead of extname:

````
irb(main):001:0> filename = "pegasus/sites.v3/code.org/public/advocacy/sample.haml.makepdf"
=> "pegasus/sites.v3/code.org/public/advocacy/sample.haml.makepdf"
irb(main):002:0> File.basename(filename, '.makepdf')
=> "sample.haml"
irb(main):003:0> File.dirname(filename)
=> "pegasus/sites.v3/code.org/public/advocacy"
irb(main):004:0> File.basename(filename, '.*')
=> "sample.haml"
irb(main):005:0> File.basename(File.basename(filename, '.makepdf'), '.*')
=> "sample"
````